src/Controls/src/Core/Platform/Windows/CollectionView/ItemContentControl.cs

```diff
@@ -263,58 +263,43 @@ void UpdateSemanticProperties(IView view)
 			this.SetAutomationPropertiesAccessibilityView(_visualElement, defaultAccessibilityView);
 		}
 
-		protected override WSize MeasureOverride(WSize availableSize)
+		/// <inheritdoc/>
+		protected override WSize ArrangeOverride(WSize finalSize)
 		{
 			if (_renderer == null)
 			{
-				return base.MeasureOverride(availableSize);
+				return base.ArrangeOverride(finalSize);
 			}
 
-			var frameworkElement = Content as FrameworkElement;
-			var formsElement = _renderer.VirtualView as VisualElement;
-			var margin = _renderer.VirtualView.Margin;
-
-			if (ItemHeight != default || ItemWidth != default)
-			{
-				formsElement.Layout(new Rect(0, 0, ItemWidth, ItemHeight));
-
-				var wsize = new WSize(ItemWidth, ItemHeight);
+			var width = ItemWidth == default ? finalSize.Width : ItemWidth;
+			var height = ItemHeight == default ? finalSize.Height : ItemHeight;
+			var size = new WSize(width, height);
 
-				frameworkElement.Margin =
-					WinUIHelpers.CreateThickness(
-						margin.Left + ItemSpacing.Left,
-						margin.Top + ItemSpacing.Top,
-						margin.Right + ItemSpacing.Right,
-						margin.Bottom + ItemSpacing.Bottom);
+			base.ArrangeOverride(size);
 
-				if (CanMeasureContent(frameworkElement))
-					frameworkElement.Measure(wsize);
+			return size;
```

Comment:
Curious why we are returning this `size`. Shouldn't we return the `base.ArrangeOverride(size)` since we're calling it in a line above?

Reply:
Yeah, that's wrong. Good catch, I'll push the fix together with the windows resizing one